junit-runtime/src/main/scala/org/junit/Assert.scala

```diff
@@ -55,7 +55,7 @@ object Assert {
 
   @inline
   private def isEquals(expected: Any, actual: Any): Boolean =
-    expected.equals(actual)
+    expected == actual
```

Comment:
This is not a correct alternative. The semantics of JUnit assertions is to use Java's equality; not Scala's cooperative equality.

We can use `java.util.Objects.equals(expected, actual)` here.